### PR TITLE
feat(argo-rollouts): support setting pod annotations separately on controller and dashboard

### PR DESCRIPTION
**Pull request by ngoduykhanh**
Hello 👋 
The current "global" `podAnnotations` config adds annotations to all pods, including the Dashboard if it is enabled. This PR gives us the ability to add pod annotations to controller & dashboard deployment separately.

The idea of this PR is similar to the one in ArgoCD's chart, in the future we probably have `global` section to include other configs as well if needed.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution! LGTM.

**Comment by ngoduykhanh** [comment]
Thanks for your review @yu-croco. 
It seems like I need one more approval to get this PR merged 🤔

**Comment by yu-croco** [comment]
There is no assigned maintainer for argo-rollouts.... 😇 
image

**Review by jmeridth — APPROVED** [review]

**Pull request merged**
